### PR TITLE
Add documentation for hydra.job.override_dirname

### PR DESCRIPTION
**Pull request by jieru-hu**
addressed #1384 
- add an example app for the config
- update example to show sweep instead of single run.

### CODE REVIEW

File: tests/test_examples/test_configure_hydra.py [hunk 1]
```diff
@@ -69,6 +69,19 @@ def test_job_name_with_config_override(tmpdir: Path) -> None:
     assert result == "name_from_config_file"
 
 
+def test_job_override_dirname(tmpdir: Path) -> None:
+    cmd = [
+        "examples/configure_hydra/job_override_dirname/my_app.py",
+        "hydra.sweep.dir=" + str(tmpdir),
+        "a=10",
+        "b=20",
```
Comment (omry) [inline tests/test_examples/test_configure_hydra.py:77]: Do a sweep over multiple runs:
```suggestion
        "a=a1,a2",
        "b=b1",
```

File: website/docs/configure_hydra/job.md [hunk 2]
```diff
@@ -95,14 +102,20 @@ hydra:
...
 With this configuration, running
 ```bash
-$ python foo.py a=10 b=20 seed=999
+$ python my_app.py --multirun a=10 b=20 seed=777,999
 ```
 
 Would result in a directory like:
 ```
-output/a=10,b=20/seed=999
+$ tree multirun/
```
Comment (omry) [inline website/docs/configure_hydra/job.md:110]: use `tree multirun -d` to list directories only.

File: website/docs/configure_hydra/job.md [hunk 3]
```diff
@@ -95,14 +102,20 @@ hydra:
 ```
 With this configuration, running
 ```bash
-$ python foo.py a=10 b=20 seed=999
+$ python my_app.py --multirun a=10 b=20 seed=777,999
```
Comment (omry) [inline website/docs/configure_hydra/job.md:105]: sweep over `a=a1,a2 b=b1 seed=1,2` to show more useful output.

File: website/docs/configure_hydra/job.md [hunk 4]
```diff
@@ -95,14 +102,20 @@ hydra:
 ```
 With this configuration, running
 ```bash
-$ python foo.py a=10 b=20 seed=999
+$ python my_app.py --multirun a=a1,a2 b=b1 seed=1,2
 ```
 
 Would result in a directory like:
```
Comment (omry) [inline website/docs/configure_hydra/job.md:108]:
```suggestion
Would result in a directory structure like:
```

**Review by omry — COMMENTED** [review]
Taking a look at the docs, I think a better place for this content is a dedicated section in https://hydra.cc/docs/next/configure_hydra/workdir, and a minimal description in the page you are currently editing.

Mine trying to refactor it a bit?

File: website/docs/configure_hydra/job.md (hunk 4 above)
Comment (jieru-hu) [reply website/docs/configure_hydra/job.md:108]: will do!

File: examples/configure_hydra/workdir/conf/config.yaml [hunk 5]
```diff
@@ -5,3 +5,5 @@ hydra:
   sweep:
```
Comment (omry) [inline examples/configure_hydra/workdir/conf/config.yaml:5]: now that the bulk of the docs are no longer under configure_hydra, move the example app accordingly.

File: website/docs/configure_hydra/job.md [hunk 6]
```diff
@@ -63,46 +63,8 @@ It is normally derived from the Python file name (The job name of the file `trai
...
-$ python foo.py a=10 b=20 seed=999
-```
-
-Would result in a directory like:
-```
-output/a=10,b=20/seed=999
-```
-Allowing you to more easily group identical runs with different random seeds together.
+This config enables creating working dir based on command line overrides. 
+For more information, check [here](/configure_hydra/workdir.md).
```
Comment (omry) [inline website/docs/configure_hydra/job.md:67]: Phrasing is a but clunky.
```suggestion
Enables the creation of an output directory which is based on command line overrides.
See [this](/configure_hydra/workdir.md) for more infromation.
```

File: website/docs/configure_hydra/workdir.md [hunk 7]
```diff
@@ -36,16 +31,85 @@ Run output directory can contain user configuration variables:
...
 
-Run output directory can contain override parameters for the job:
+then run the application with command-line overrides:
 
-See [Override dirname](./job#hydrajoboverride_dirname) in the Job configuration page for details on how to customize
-`hydra.job.override_dirname`.
+```bash
+python my_app.py --multirun a=a1 b=b1
+```
+Would result in creating a output dir `multirun/a=a1,b=b1`
```
Comment (omry) [inline website/docs/configure_hydra/workdir.md:80]: This example is a bit confusing. let's try to use something more concrete in the the example and the docs.

Maybe this:

```
python my_app.py --multirun db=mysql schema=school
```

Also, why use multirun if you are only running a single job?
seems like the example would work better with:
```
python my_app.py --multirun db=mysql schema=school,support
```

**Review by omry — CHANGES_REQUESTED** [review]
marking as request changes.

File: examples/configure_hydra/workdir/conf/config.yaml (hunk 5 above)
Comment (jieru-hu) [reply examples/configure_hydra/workdir/conf/config.yaml:5]: Not sure what you mean here 🤔 the docs are still under configure_hydra.

File: website/docs/configure_hydra/workdir.md (hunk 7 above)
Comment (jieru-hu) [reply website/docs/configure_hydra/workdir.md]: sure, updated the example.

File: website/docs/configure_hydra/job.md (hunk 6 above)
Comment (jieru-hu) [reply website/docs/configure_hydra/job.md]: fixed

File: examples/configure_hydra/job_override_dirname/config.yaml [hunk 8]
```diff
@@ -8,6 +8,6 @@ hydra:
         exclude_keys:
           - seed
 
-a: a1
-b: b1
+lr: 0.1
+optimizer: adam
```
Comment (omry) [inline examples/configure_hydra/job_override_dirname/config.yaml:12]: optimizer is typically a group.
How about:
```yaml
learning_rate: 0.1
batch_size: 32
```

File: examples/configure_hydra/workdir/conf/config.yaml (hunk 5 above)
Comment (omry) [reply examples/configure_hydra/workdir/conf/config.yaml:5]: got confused. looks okay.

File: tests/test_examples/test_configure_hydra.py [hunk 9]
```diff
@@ -69,6 +69,20 @@ def test_job_name_with_config_override(tmpdir: Path) -> None:
...
+        "examples/configure_hydra/job_override_dirname/my_app.py",
+        "hydra.sweep.dir=" + str(tmpdir),
+        "lr=0.1,0.01",
+        "optimizer=adam",
+        "seed=999",
+        "-m",
+    ]
+    run_python_script(cmd)
+    assert Path(tmpdir / "lr=0.1,optimizer=adam/seed=999/my_app.log").exists()
+    assert Path(tmpdir / "lr=0.01,optimizer=adam/seed=999/my_app.log").exists()
```
Comment (omry) [inline tests/test_examples/test_configure_hydra.py:83]: Why not directly check that the dir exists instead of a file in it?

File: examples/configure_hydra/workdir/conf/config.yaml [hunk 10]
```diff
@@ -5,3 +5,5 @@ hydra:
...
     dir: sweep_dir
     subdir: ${hydra.job.num}
+
+a: a1
```
Comment (omry) [inline examples/configure_hydra/workdir/conf/config.yaml:9]: Why change this?

File: website/docs/configure_hydra/workdir.md [hunk 11]
```diff
@@ -36,16 +31,91 @@ Run output directory can contain user configuration variables:
...
+
+### Configuration for multirun
 
+Sweep sub directory contains the job number for the job instance:
+```yaml
+hydra:
+  sweep:
+    dir: sweep_dir
+    subdir: ${hydra.job.num}
+```
```
Comment (omry) [inline website/docs/configure_hydra/workdir.md:44]: This is almost the default, except the dir does not have the timestamp.
So what you are actually doing here is to prevent the timestamp. Try to clarify what you are trying to demonstrate before adding examples.

File: examples/configure_hydra/job_override_dirname/config.yaml (hunk 8 above)
Comment (jieru-hu) [reply examples/configure_hydra/job_override_dirname/config.yaml]: sure, updated.

File: examples/configure_hydra/workdir/conf/config.yaml (hunk 10 above)
Comment (jieru-hu) [reply examples/configure_hydra/workdir/conf/config.yaml:9]: in the corresponding example doc, I want to do `python my_app.py a=a1,a2,a3` without the `+` to make it cleaner.

File: tests/test_examples/test_configure_hydra.py (hunk 9 above)
Comment (jieru-hu) [reply tests/test_examples/test_configure_hydra.py]: sure, changed to check dir.

File: website/docs/configure_hydra/workdir.md (hunk 11 above)
Comment (jieru-hu) [reply website/docs/configure_hydra/workdir.md:44]: Before this yaml the doc says `Sweep sub directory contains the job number for the job instance. `

Are you saying we should add more context here?

File: examples/configure_hydra/workdir/conf/config.yaml (hunk 10 above)
Comment (omry) [reply examples/configure_hydra/workdir/conf/config.yaml:9]: Gotcha.

File: website/docs/configure_hydra/workdir.md (hunk 11 above)
Comment (omry) [reply website/docs/configure_hydra/workdir.md:44]: I am saying that you should make up your mind about what you are demonstrating.
In the code, you are overriding hydra.sweep.dir but keeping the subdir as the default.
It seems odd given that your sentence is talking about the sub dir.

Maybe open by describing the default behavior and then show a few ways to change it.

Comment (jieru-hu) [reply website/docs/configure_hydra/workdir.md]: got it! sorry I misunderstood you earlier.

File: website/docs/configure_hydra/workdir.md [hunk 12]
```diff
@@ -36,16 +31,102 @@ Run output directory can contain user configuration variables:
...
+        └── 2
+```
+Similar configuration patterns in run can be applied to config multirun dir as well.
+
+For example, multirun output directory grouped by job name, and sub dir by job num:
+```yaml
+hydra:
+  sweep:
+    dir: ${hydra.job.name}
+    subdir: ${hydra.job.num}
```
Comment (omry) [inline website/docs/configure_hydra/workdir.md:67]: Show the tree structure. this is a bit confusing (job.name, job.num).

Also, you can use column layout to put the config box next to the tree conf.
That would make things more compact.

(Do it only if you can make the two columns fit nicely without a scroll bar).

File: website/docs/configure_hydra/workdir.md [hunk 13]
```diff
@@ -36,16 +31,102 @@ Run output directory can contain user configuration variables:
...
+
+### Using `hydra.job.override_dirname`
+
+<ExampleGithubLink text="Example application" to="examples/configure_hydra/job_override_dirname"/>
+
+This field is populated automatically using your command line arguments and is typically being used as a part of your 
+output directory pattern. It is meant to be used along with the configuration for working dir, especially
+in `hydra.sweep.subdir`.
+
+For example, we configure the example application like the following
```
Comment (omry) [inline website/docs/configure_hydra/workdir.md:78]:
```suggestion
As an example, using this configuration:
```

File: website/docs/configure_hydra/workdir.md [hunk 14]
```diff
@@ -36,16 +31,102 @@ Run output directory can contain user configuration variables:
...
+```yaml
+hydra:
+  sweep:
+    dir: multirun
+    subdir: ${hydra.job.override_dirname}
+```
+
+then run the application with command-line overrides:
```
Comment (omry) [inline website/docs/configure_hydra/workdir.md:86]:
```suggestion
Will result in this output directory structure:
```

File: website/docs/configure_hydra/workdir.md [hunk 15]
```diff
@@ -36,16 +31,102 @@ Run output directory can contain user configuration variables:
...
 
+```bash
+python my_app.py --multirun batch_size=32 learning_rate=0.1,0.01
 ```
+Would result in creating the following output structure
+```bash
+$ tree multirun -d
+multirun
+├── batch_size=32,learning_rate=0.01
+└── batch_size=32,learning_rate=0.1
```
Comment (omry) [inline website/docs/configure_hydra/workdir.md:96]: If this does not fit in a two column layout next to the corresponding config you can use different key names.

**Review by omry — CHANGES_REQUESTED** [review]

File: website/docs/configure_hydra/workdir.md (hunk 12 above)
Comment (jieru-hu) [reply website/docs/configure_hydra/workdir.md:79]: Yea, good idea. Created columns to fix the configs :)

File: website/docs/configure_hydra/workdir.md (hunk 15 above)
Comment (jieru-hu) [reply website/docs/configure_hydra/workdir.md:127]: Fixed :)

**Review by omry — APPROVED** [review]
looking good.
